sklearn/cluster/_optics.py

```diff
@@ -243,7 +245,15 @@ def fit(self, X, y=None):
         self : instance of OPTICS
             The instance.
         """
-        X = self._validate_data(X, dtype=float)
+
+        dtype = bool if self.metric in PAIRWISE_BOOLEAN_FUNCTIONS else float
+        if dtype == bool and X.dtype != bool:
+            msg = f"Data will be converted to boolean for" \
+                  f" metric {self.metric}, to avoid this warning," \
+                  f" you may convert the data prior to calling fit."
```

Comment:
we try to avoid backslashes

```suggestion
            msg = (f"Data will be converted to boolean for"
                   f" metric {self.metric}, to avoid this warning,"
                   f" you may convert the data prior to calling fit.")
```